### ISSUE TITLE
Comparing objects with empty prototypes (no hasOwnProperty method)

### ISSUE THREAD

**Issue by denisnd** [mattphillips/deep-object-diff#7]
I'm using deep-object-diff to compare objects generated by [query-string](https://github.com/sindresorhus/query-string). Structures returned by its parser have no hasOwnProperty method (https://github.com/sindresorhus/query-string/issues/47) and so deep-object-diff can't work on them as it relies on the method.

Steps to reproduce:
```
var deepObjectDiff = require('deep-object-diff');

var a = Object.create(null);
a.foo = 'bar;

var b = Object.create(null);
b.foo = 'baz';

console.log( deepObjectDiff.diff(a, b) );
```

Error message:

```
TypeError: rhs.hasOwnProperty is not a function
    at C:\Users\nd\object-diff\node_modules\deep-object-diff\dist\diff\index.js:55:18
```

**Comment by mattphillips** [comment]
@denisnd Hey thanks for raising the issue! I've just published a fix in version [1.0.3](https://www.npmjs.com/package/deep-object-diff) so if you try the same code again all should be good :smile:.

It will treat the object as a normal JS object that has the `.hasOwnProperty` function available, this means that it will work on nested structures that are missing this property at any level.

```js
var deepObjectDiff = require('deep-object-diff');

var a = Object.create(null);
a.foo = 'bar';

var b = Object.create(null);
b.foo = 'baz';

console.log( deepObjectDiff.diff(a, b) ); // { foo: 'baz' }
```

**Comment by denisnd** [comment]
@mattphillips Thank you for quick fixing! Now everything is working fine.
